routes/paths.js

```diff
@@ -3,6 +3,184 @@ const express = require('express');
+		let points = await getPoints(curNode, nextNode);
+		for (let p of points) {
+			locations.unshift(p);
+		}
+		locations.unshift(nextNode.location);
+		curNode = nextNode;
+	}
+	return locations;
+};
+
+const getPoints = async (node1, node2) => {
+	let path = await Path.findOne({
+		nodeA: node1,
+		nodeB: node2,
+	});
+	let points = [];
+	//console.log(path);
+	if (path === null) {
+		path = await Path.findOne({
+			nodeA: node2,
+			nodeB: node1,
+		});
+		points = path.points.reverse();
+	} else {
+		points = path.points;
+	}
+	//console.log(path);
+	return points;
+};
+
+mapRouter.route('/pathBetween').get(async (req, res) => {
+	let startNode = await MapNode.findOne({
+		'location.latitude': req.body.startLat,
+		'location.longitude': req.body.startLon,
+	});
+	if (startNode === null) {
+		startNode = await getClosestMapNode(req.body.startLat, req.body.startLon);
+	}
+	let endNode = await MapNode.findOne({
+		'location.latitude': req.body.endLat,
+		'location.longitude': req.body.endLon,
+	});
+	if (endNode === null) {
+		endNode = await getClosestMapNode(req.body.endLat, req.body.endLon);
+	}
+	const nodes = await getPathBetween(startNode, endNode);
+	//console.log(nodes);
+	res.json(nodes);
+});
+
+const alreadyHas = (node, map) => {
+	//console.log(node)
+	//console.log(set);
+	for (let n of map.keys()) {
+		if (
+			node.location.latitude === n.location.latitude &&
+			node.location.longitude === n.location.longitude
+		) {
+			return true;
+		}
+	}
+	return false;
+};
+
+const getPathBetween = async (start, end) => {
+	let gScore = new Map();
+	let fScore = new Map();
+	let cameFrom = new Map();
+	let queue = new Set();
+
+	gScore.set(start, 0);
+	fScore.set(start, heuristic(start, end));
+	queue.add(start);
+
+	while (queue.size !== 0) {
+		let curNode;
+		let smallestF = Number.MAX_SAFE_INTEGER;
+
+		for (let node of queue) {
+			//console.log(fScore.get(node));
+			if (fScore.get(node) < smallestF) {
+				curNode = node;
+				smallestF = fScore.get(node);
+			}
+		}
+		//console.log(curNode);
+		queue.delete(curNode);
+		if (
+			curNode.location.latitude === end.location.latitude &&
+			curNode.location.longitude === end.location.longitude
+		) {
+			return await reconstructPath(cameFrom, curNode);
+		}
+		let neighbors = await getNeighbors(curNode);
+		for (let neighbor of neighbors) {
+			const points = await getPoints(curNode, neighbor);
+			let dist = points.length + 1;
+			//console.log(dist)
+			const tempG = dist + gScore.get(curNode);
+			const beenVisited = alreadyHas(neighbor, gScore);
+			if (!beenVisited || tempG < gScore.get(neighbor)) {
+				if (!beenVisited) {
+					queue.add(neighbor);
+				}
+				gScore.set(neighbor, tempG);
+				fScore.set(neighbor, tempG + heuristic(neighbor, end));
+				cameFrom.set(neighbor, curNode);
+			}
+		}
+	}
+	return 'No Path Found';
+};
+
+mapRouter.route('/resetPath').delete(async (req, res) => {
+	await Path.deleteMany({});
+	await MapNode.deleteMany({});
+	res.json('done');
+});
```

Comment:
All commits are pushed, and we def don't want this in prod, so remove this. You can easily drop the Path and MapNode collection through the MongoDB Atlas web console, or you can download MongoDB Compass and login through the URI as well.